src/cephadm/cephadm

```diff
@@ -5212,6 +5257,462 @@ def command_gather_facts():
 ##################################
 
 
+class CephadmDaemonHandler(BaseHTTPRequestHandler):
+    api_version = 'v1'
+    valid_routes = [
+        f'/{api_version}/metadata',
+        f'/{api_version}/metadata/health',
+        f'/{api_version}/metadata/disks',
+        f'/{api_version}/metadata/daemons',
+        f'/{api_version}/metadata/host',
+    ]
+    
+    def _help_page(self):
+        return """<!DOCTYPE html>
+<html>
+<head><title>cephadm metadata exporter</title></head>
+<style>
+body {{
+  font-family: sans-serif;
+  font-size: 0.8em;
+}}
+table {{
+  border-width: 0px;
+  border-spacing: 0px;
+  margin-left:20px;
+}}
+tr:hover {{
+  background: PowderBlue;
+}}
+td,th {{
+  padding: 5px;
+}}
+</style>
+<body>
+    <h1>cephadm metadata exporter {api_version}</h1>
+    <table>
+      <thead>
+        <tr><th>Endpoint</th><th>Methods</th><th>Response</th><th>Description</th></tr>
+      </thead>
+      <tr><td><a href='{api_version}/metadata'>{api_version}/metadata</a></td><td>GET</td><td>JSON</td><td>Return <b>all</b> metadata for the host</td></tr>
+      <tr><td><a href='{api_version}/metadata/daemons'>{api_version}/metadata/daemons</a></td><td>GET</td><td>JSON</td><td>Return daemon and systemd states for ceph daemons</td></tr>
+      <tr><td><a href='{api_version}/metadata/disks'>{api_version}/metadata/disks</a></td><td>GET</td><td>JSON</td><td>show disk inventory (ceph-volume)</td></tr>
+      <tr><td><a href='{api_version}/metadata/health'>{api_version}/metadata/health</a></td><td>GET</td><td>JSON</td><td>Show current health of the exporter sub-tasks</td></tr>
+      <tr><td><a href='{api_version}/metadata/host'>{api_version}/metadata/host</a></td><td>GET</td><td>JSON</td><td>Show host metadata</td></tr>
+    </table>
+</body>
+</html>""".format(api_version=CephadmDaemonHandler.api_version)
+
+    def _fetch_root(self):
+        self.send_response(200)
```

Comment:
Please add conditional returncodes

Reply:
conditional on what? this is a request to '/' which returns static content - what is the conditional check that you want to add?